cmd/internal/persist_output.go

```diff
@@ -22,10 +27,47 @@ into a versionsed secret or kube native secret using flags specified to this com
 			return errors.Errorf("persist-output command failed. cf-operator-namespace flag is empty.")
 		}
 
-		return extendedjob.PersistOutput(namespace)
+		// hostname of the container is the pod name in kubernetes
+		podName, err := os.Hostname()
```

Comment:
Is it safe to get pod name? Could we use downward API and inject env?

```
spec:
  containers:
    - name: test-container
      image: gcr.io/google_containers/busybox
      command: [ "/bin/sh", "-c", "env" ]
      env:
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
```

Reply:
The kubernetes docs say that pod name available by hostname. [0]

[0] https://kubernetes.io/docs/concepts/containers/container-environment-variables/#container-information